### PR TITLE
[NFS-Ganesha][Automation]Rootsquash enable using conf file

### PR DESCRIPTION
**Pull request by MohitBis**
[NFS-Ganesha][Automation]Rootsquash enable using conf file

### CODE REVIEW

**Comment by MohitBis** [comment]
Logs : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-FJFKHE/

**Comment by MohitBis** [comment]
Latest Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-F2DB98/

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Review by msainiRedhat — APPROVED** [review]
LGTM

**Review by msainiRedhat — DISMISSED** [review]
LGTM

File: cli/utilities/utils.py [hunk 1]
```diff
@@ -678,3 +678,26 @@ def generate_random_string(**kw):
...
+        filepath (str): file path
+        clients (ceph): Client nodes
+    """
+    out = clients[0].exec_command(sudo=True, cmd=f"ls -n {filepath}")
+    uid = int(out[0].split()[2])
+    gid = int(out[0].split()[3])
+    if uid == 0 and gid == 0:
+        log.info(f"The file '{filepath}' is created by the root user.")
+        return "rootuser"
+    elif uid == 4294967294 and gid == 4294967294:
```
Comment (vamahaja) [inline cli/utilities/utils.py:696]: Add description for gid & uid numbers used

File: tests/nfs/nfs_rootsquash_using_conf.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            cmd=f"ceph nfs export get {nfs_name} {nfs_export_squash} > export.conf",
+        )
+
+        client.exec_command(
+            sudo=True,
+            cmd=f"sed -i 's/{original_squash_value}/{new_squash_value}/g' export.conf",
+        )
+        client.exec_command(
+            sudo=True,
+            cmd=f"ceph nfs export apply {nfs_name} -i export.conf",
```
Comment (vamahaja) [inline tests/nfs/nfs_rootsquash_using_conf.py:27]: Add library

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Comment by vamahaja** [comment]
Rebase PR

File: cli/utilities/utils.py (hunk 1 above)
Comment (MohitBis) [reply cli/utilities/utils.py:698]: Updated

File: tests/nfs/nfs_rootsquash_using_conf.py (hunk 2 above)
Comment (MohitBis) [reply tests/nfs/nfs_rootsquash_using_conf.py]: Updated

**Comment by MohitBis** [comment]
Latest Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-SGP2QU/

**Review by vamahaja — CHANGES_REQUESTED** [review]

File: cli/ceph/nfs/export/export.py [hunk 3]
```diff
@@ -51,3 +51,15 @@ def delete(self, cluster, export):
...
             return out[0].strip()
         return out
+
+    def apply(self, nfs_name):
+        """
+        apply a given nfs export
+        Args:
+            nfs_name (str): Name of nfs cluster
+        """
+        cmd = f"{self.base_cmd} apply {nfs_name} -i export.conf"
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply cli/ceph/nfs/export/export.py]: Updated

File: tests/nfs/nfs_rootsquash_using_conf.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+log = Log(__name__)
+
+
+def rootsquash_using_conf(
+    client, nfs_name, nfs_export_squash, original_squash_value, new_squash_value
+):
+    try:
+        client.exec_command(
+            sudo=True,
+            cmd=f"ceph nfs export get {nfs_name} {nfs_export_squash} > export.conf",
```
[the remark this answers is not in the document]
Comment (MohitBis) [reply tests/nfs/nfs_rootsquash_using_conf.py]: Updated

**Comment by MohitBis** [comment]
Latest Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GIKNXM/

**Review by pranavprakash20 — APPROVED** [review]
LGTM

File: cli/ceph/nfs/export/export.py [hunk 5]
```diff
@@ -51,3 +51,30 @@ def delete(self, cluster, export):
...
+
+    def get(self, nfs_name, nfs_export, export_conf):
+        """
+        get a given nfs export conf
+        Args:
+            nfs_name (str): Name of nfs cluster
+            nfs_export (str): Name of nfs export
+            export_conf (str): Export conf file name
+        """
+        cmd = f"{self.base_cmd} get {nfs_name} {nfs_export} > {export_conf}"
```
Comment (vamahaja) [inline cli/ceph/nfs/export/export.py:63]: Don't direct `export_conf`, execute command and then create `export_conf`

File: tests/nfs/nfs_rootsquash_using_conf.py [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+from cli.utilities.utils import get_file_owner
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def rootsquash_using_conf(
+    client, nfs_name, nfs_export_squash, original_squash_value, new_squash_value
+):
+    try:
```
Comment (vamahaja) [inline tests/nfs/nfs_rootsquash_using_conf.py:15]: Add description for method

**Review by vamahaja — CHANGES_REQUESTED** [review]

**Review by pranavprakash20 — APPROVED** [review]
LGTM

**Review by msainiRedhat — APPROVED** [review]
LGTM

**Review by vdas-redhat — DISMISSED** [review]

**Comment by MohitBis** [comment]
Latest Log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-BN3EZP/

**Review by vamahaja — APPROVED** [review]

**Pull request merged**